### PR TITLE
[Dubbo-2348 ] AbstractRegistry util test enhancement

### LINKED ISSUE #2348: AbstractRegistry unit test enhancement.
Design and provide unit tests to cover more use cases.

### PR DESCRIPTION
**Pull request by lllichen**
## What is the purpose of the change

fix #2348 

## Brief changelog

add AbstractRegistryTest getCacheUrlsTest method

## Verifying this change

XXXXX

Follow this checklist to help us incorporate your contribution quickly and easily:

- [x] Make sure there is a [GITHUB_issue](https://github.com/apache/incubator-dubbo/issues) filed for the change (usually before you start working on it). Trivial changes like typos do not require a GITHUB issue. Your pull request should address just this issue, without pulling in other changes - one PR resolves one issue.
- [x] Format the pull request title like `[Dubbo-XXX] Fix UnknownException when host config not exist #XXX`. Each commit in the pull request should have a meaningful subject line and body.
- [x] Write a pull request description that is detailed enough to understand what the pull request does, how, and why.
- [x] Write necessary unit-test to verify your logic correction, more mock a little better when cross module dependency exist. If the new feature or significant change is committed, please remember to add integration-test in [test module](https://github.com/apache/incubator-dubbo/tree/master/dubbo-test).
- [ ] Run `mvn clean install -DskipTests` & `mvn clean test-compile failsafe:integration-test` to make sure unit-test and integration-test pass.
- [ ] If this contribution is large, please follow the [Software Donation Guide](https://github.com/apache/incubator-dubbo/wiki/Software-donation-guide).

### CODE REVIEW

File: dubbo-registry/dubbo-registry-api/src/test/java/org/apache/dubbo/registry/support/AbstractRegistryTest.java [hunk 1]
```diff
@@ -48,6 +49,8 @@
     @Before
     public void init() {
         URL url = URL.valueOf("dubbo://" + NetUtils.getLocalAddress().getHostName() + ":2233");
+        //sync update cache file
+        url = url.addParameter( "save.file",true );
```
Comment (carryxyh) [inline dubbo-registry/dubbo-registry-api/src/test/java/org/apache/dubbo/registry/support/AbstractRegistryTest.java]: not the same code style.
`use ("save.file", true);`

File: dubbo-registry/dubbo-registry-api/src/test/java/org/apache/dubbo/registry/support/AbstractRegistryTest.java [hunk 2]
```diff
@@ -515,4 +518,19 @@ public void allTest() throws Exception {
...
+        urls.add(testUrl);
+        // check if notify successfully
+        Assert.assertFalse(notifySuccess);
+        abstractRegistry.notify(testUrl, listener, urls);
+        Assert.assertTrue(notifySuccess);
+        List<URL> cacheUrl = abstractRegistry.getCacheUrls( testUrl );
+        Assert.assertTrue( cacheUrl.size() == 1 );
+        URL nullUrl = URL.valueOf("http://1.2.3.4:9090/registry?check=false&file=N/A&interface=com.testa");
+        cacheUrl = abstractRegistry.getCacheUrls( nullUrl );
+        Assert.assertTrue( Objects.isNull( cacheUrl ) );
```
Comment (carryxyh) [inline dubbo-registry/dubbo-registry-api/src/test/java/org/apache/dubbo/registry/support/AbstractRegistryTest.java]: Here, too.

File: dubbo-registry/dubbo-registry-api/src/test/java/org/apache/dubbo/registry/support/AbstractRegistryTest.java (hunk 1 above)
Comment (lllichen) [reply dubbo-registry/dubbo-registry-api/src/test/java/org/apache/dubbo/registry/support/AbstractRegistryTest.java]: ok

File: dubbo-registry/dubbo-registry-api/src/test/java/org/apache/dubbo/registry/support/AbstractRegistryTest.java (hunk 2 above)
Comment (lllichen) [reply dubbo-registry/dubbo-registry-api/src/test/java/org/apache/dubbo/registry/support/AbstractRegistryTest.java]: ok

**Comment by carryxyh** [comment]
Hi, pls resolve conflicts.

File: dubbo-registry/dubbo-registry-api/src/test/java/org/apache/dubbo/registry/support/AbstractRegistryTest.java [hunk 3]
```diff
@@ -48,6 +49,8 @@
     @Before
     public void init() {
         URL url = URL.valueOf("dubbo://" + NetUtils.getLocalAddress().getHostName() + ":2233");
+        //sync update cache file
+        url = url.addParameter("save.file",true);
```
Comment (carryxyh) [inline dubbo-registry/dubbo-registry-api/src/test/java/org/apache/dubbo/registry/support/AbstractRegistryTest.java]: Hi, format your code, pls.

**Comment by carryxyh** [comment]
I am very sorry that the processing is not timely. 
Please resolve the conflicts and format code, then I will process your pr.

**Comment by lllichen** [comment]
@carryxyh i have fixed the conflicts ,and waiting the ci pogress

**Comment by carryxyh** [comment]
Of course you can. :)
LGTM.

**Pull request merged**